src/types.jl

```diff
@@ -81,7 +81,21 @@ function FrameCode(scope, src::CodeInfo; generator=false, optimize=true)
         end
     end
     used = find_used(src)
-    return FrameCode(scope, src, methodtables, breakpoints, used, generator)
+    framecode = FrameCode(scope, src, methodtables, breakpoints, used, generator)
+    if scope isa Method && !is_generated(scope) # TODO: Relax this
+        method = scope
+        for bp in unresolved_breakpoints
+            method.file === bp.file || continue
+            if method_contains_line(method, bp.line)
+                stmtidx = statementnumber(framecode, bp.line)
+                breakpoint!(framecode, stmtidx, bp.condition)
+            end
+        end
+        lineranges = get!(() -> Pair{UnitRange, FrameCode}[], framecode_locations, method.file)
```

Comment:
This line and the next need a comment, it took me a bit to understand what this was about.